File: web3rpc/rpc-specs/code-samples/python/debug/profiling/startPProf.py
import json
from web3 import Web3
from web3py_ext import extend

host = "https://api.baobab.klaytn.net:8651"

address = "localhost"
port = 6060

w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.start_p_prof(address, port)

print(json.loads(debug_response.response.data))

File: web3rpc/rpc-specs/code-samples/javascript/debug/profilling/startPProf.js
const { JsonRpcProvider } = require("@klaytn/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-baobab.klaytn.net");
  const address = "localhost";
  const port = 6060;

  provider.debug
    .startPProf({ address, port }, (err, data, response) => {})
    .then((data) => {
      console.log(data);
    });
})();
